spotinst_sdk2/models/ocean/aws/__init__.py

```diff
@@ -1,370 +1,918 @@
+            is_enabled: bool = none,
+            parameters: Parameters = none,
+            task_type: TaskType = none):
+        self.cron_expression = cron_expression
+        self.is_enabled = is_enabled
+        self.parameters = parameters
+        self.task_type = task_type
+
+
+class ShutdownHours:
+    """
+    # Arguments
+    is_enabled: bool
+    time_windows: List[str]
+    """
+    def __init__(
+            self,
+            is_enabled: bool = none,
+            time_windows: List[str] = none):
+        self.is_enabled = is_enabled
+        self.time_windows = time_windows
+
+
+class Scheduling:
+    """
+    # Arguments
+    tasks: List[Tasks]
+    shutdown_hours: ShutdownHours
+    """
+    def __init__(
+            self,
+            tasks: List[Tasks] = none,
+            shutdown_hours: ShutdownHours = none):
+        self.tasks = tasks
+        self.shutdown_hours = shutdown_hours
+# endregion
+
+
+# region Security
+class ContainerImage:
+    """
+    # Arguments
+    approved_images: List[str]
+    """
+    def __init__(
+            self,
+            approved_images: List[str] = none):
+        self.approved_images = approved_images
+
+
+class Security:
+    """
+    # Arguments
+    container_image: ContainerImage
+    """
+    def __init__(
+            self,
+            container_image: ContainerImage = none):
+        self.container_image = container_image
+# endregion
+
+
 # region Compute
-class Compute:
-	"""
-	# Arguments
-	instance_types: InstanceTypes
-	subnet_ids: List[str]
-	launch_specification: LaunchSpecifications
-	"""
-	def __init__(
-		self,
-		instance_types=none,
-		subnet_ids=none,
-		launch_specification=none):
-
-		self.instance_types = instance_types
-		self.subnet_ids = subnet_ids
-		self.launch_specification = launch_specification
+# region InstanceTypes
+class Architectures(Enum):
+    i386 = "i386"
+    x86_64 = "x86_64"
+    arm64 = "arm64"
+
+
+class Categories(Enum):
+    accelerated_computing = "Accelerated_computing"
+    compute_optimized = "Compute_optimized"
+    general_purpose = "General_purpose"
+    memory_optimized = "Memory_optimized"
+    storage_optimized = "Storage_optimized"
+
+
+class DiskTypes(Enum):
+    nvme = "NVMe"
+    ebs = "EBS"
+    ssd = "SSD"
+    hdd = "HDD"
+
+
+class Hypervisor:
+    nitro = "nitro"
+    xen = "xen"
+
+
+class RootDeviceTypes(Enum):
+    ebs = "ebs"
+    instance_store = "instance-store"
+
+
+class VirtualizationTypes(Enum):
+    hvm = "hvm"
+    paravirtual = "paravirtual"
+
+
+class InstanceTypesFilters:
+    """
+    # Arguments
+    architectures: Architectures
+    categories: Categories
+    disk_types: DiskTypes
+    exclude_families: List[str]
+    exclude_metal: bool
+    hypervisor: Hypervisor
+    include_families: List[str]
+    is_ena_supported: bool
+    max_gpu: int
+    max_memory_giB: int
+    max_network_performance: int
+    max_vcpu: int
+    min_enis: int
+    min_gpu: int
+    min_memory_giB: int
```

Comment:
Recheck this parameter.

Reply:
minMemoryGiB | number <double> >= 0Minimum amount of Memory (GiB).

We have this parameter in docs